### PR TITLE
Country info in Project metadata

### PR DESCRIPTION
**Pull request by ramyaragupathy**
Closed the [earlier PR](https://github.com/hotosm/tasking-manager/pull/1611) due to rebase issues.

**Implementation Items:**
- [x] Setup migration to populate country info for existing projects
- [x] Frontend element in the metadata section to display existing country field
- [x] API call to fetch the country list
- [x] Upsert function to update country info
- [x] Geocoding API call to determine country with the initial geometry
- [x] Fix failing tests

image

cc @xamanu

### CODE REVIEW

**Comment by ramyaragupathy** [comment]
Unit tests fails at `test_translation_source_file_contains_all_translatables - tests.server.unit.services.test_translation_service.TestTranslationService`

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.7/unittest/case.py", line 628, in run
    testMethod()
  File "/home/circleci/app/tests/server/unit/services/test_translation_service.py", line 40, in test_translation_source_file_contains_all_translatables
    self.assertEqual(count_new_strings, 0)
  File "/usr/local/lib/python3.7/unittest/case.py", line 852, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/local/lib/python3.7/unittest/case.py", line 845, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: 3 != 0
```

Haven't touched any of the translation file in this branch. Not sure what's causing this

**Comment by ramyaragupathy** [comment]
One other issue:
- Country information was not populated for close to 1500 projects. I suspect this is related to ocean geometry from task splitting issue https://github.com/hotosm/tasking-manager/issues/1282

**Comment by ramyaragupathy** [comment]
Per voice w/ @xamanu - unit test failure fixed with `python3 manage.py refresh_translatables` to export recent strings from front end files 🎉

File: server/models/postgis/project.py [hunk 1]
```diff
@@ -135,6 +139,30 @@ def set_default_changeset_comment(self):
         self.changeset_comment = f'{default_comment}-{self.id} {self.changeset_comment}' if self.changeset_comment is not None else f'{default_comment}-{self.id}'
         self.save()
 
+    def set_country_info(self):
+        """ Sets the default country based on centroid"""
+        query = """SELECT ST_Y(ST_Transform(ST_Centroid(centroid), 4326)) || ', '
```
Comment (JorgeMartinezG) [inline server/models/postgis/project.py]: @ramyaragupathy I think we can use sqlalchemy here. Something like a static method similar to https://github.com/hotosm/tasking-manager/blob/develop/server/models/postgis/project.py#L632 :
```
lat, lng = db.session.query(func.ST_Y(Project.centroid), func.ST_X(Project.centroid)).filter(Project.id==project_id).one()
```

Comment (ramyaragupathy) [reply server/models/postgis/project.py]: Thanks for the flag @JorgeMartinezG. Replaced the centroid query and also the country fetch query

**Comment by xamanu** [comment]
Started testing this today, but then as per conversation with @ramyaragupathy we should wait with the review still.

**Comment by ramyaragupathy** [comment]
Looking to reduce the migration time (~2 hrs) by increasing the match probability using continent info

**Comment by ramyaragupathy** [comment]
> Looking to reduce the migration time (~2 hrs) by increasing the match probability using continent info

This now takes about 40 minutes

**Comment by ramyaragupathy** [comment]
Now the migration is done in 20 minutes 🎉

**Comment by thadk** [comment]
I think it would be best to adapt the React interface with this after the `/client` code is removed and the APIv2 is in here. Looks okay though

**Comment by ramyaragupathy** [comment]
Moving this to https://github.com/hotosm/tasking-manager/pull/1863

**Pull request closed without merging**